src/application/tester/tester.go

```diff
@@ -42,17 +43,22 @@ func TestApp(appContext types.AppContext, writer io.Writer, mode composebuilder.
 				return testResult, nil
 			}
 			if !testResult.Passed {
-				numFailed++
+				failedTests = append(failedTests, testRole)
 			}
 		}
 	}
-	if numFailed == 0 {
-		util.PrintSectionHeader(writer, "All tests passed\n\n")
+	if len(failedTests) == 0 {
+		green := color.New(color.FgGreen)
+		green.Fprint(writer, "All tests passed\n\n")
 		return types.TestResult{
 			Passed: true,
 		}, nil
 	}
-	util.PrintSectionHeaderf(writer, "%d tests failed\n\n", numFailed)
+	red := color.New(color.FgRed)
+	red.Fprint(writer, "The following tests failed:\n")
+	for _, failedTest := range failedTests {
+		red.Fprintln(writer, failedTest)
+	}
```

Comment:
To combat the lint failure, can we extract printing the results to its own function?